### ISSUE TITLE
Expose request body for rest (feature request)

### ISSUE THREAD

**Issue by vloth** [jihchi/res-msw#8]
It's fairly common to access the message body whenever the rest method is POST or PUT. I did a quick test changing the type of request to ` {params: Js.Dict.t<string>, body: Js.Dict.t<string>}`, but not sure if that's the best way.
Are there any plans to support this in the near future?

**Comment by jihchi** [comment]
Hi @vloth ,

According to msw's [doc](https://mswjs.io/docs/api/request#properties) and [code](https://github.com/mswjs/msw/blob/master/src/handlers/RequestHandler.ts#L21-L30), it seems to be better to make body type similar to this:

```res
type Body =
| PlainText(string)
| Record(Js.Dict.t<string>)

type request = {
  body: Js.Undefined.t<Body>
}
```

Does this work for you?

**Comment by vloth** [comment]
It does! thanks! Let me know if I can help in any way

**Comment by jihchi** [comment]
PR is welcome! Would you like to work on this and create a PR?

**Comment by vloth** [comment]
Yes! Not really knowledged in a Rescript yet, but will try

**Comment by vloth** [comment]
Just to check if I'm in the right direction, I believe to achieve a type similar to what you first proposed I'd need a new function to get the body and convert it to the type body right?
Just extending the type request with the type definitions won't be enough as rescript won't know how to properly map the string to a PlainText and the js object to Record tags.

**Comment by jihchi** [comment]
We (res-msw binding) don't need to know how to distinguish the type of `request.body`, users of binding would need to take care of it when they are accessing `request.body`, for example, it might look like this in users' code:

```res
rest |> post("https://api.github.com/repos/:owner/:repo", (
  req,
  res,
  ctx,
) => {
  let body = switch Js.Undefined.toOption(req.body) {
  | Some(PlainText(text)) => text
  | Some(Record(record)) =>
    record |> Js.Dict.get(_, "foo") |> Option.getWithDefault(_, "N/A")
  | None => ""
  }

  res |> Rest.mock([ctx |> Rest.status(200), ctx |> Rest.text(body)])
})
```

**Comment by vloth** [comment]
Hi, I'm very sorry for the radio silence! But finally got around to start some work. I spend quite some time trying to figure out a way to make the bindings for the request body variant, but couldn't find a way. The closest feature would be the [@unwrap](https://rescript-lang.org/docs/manual/latest/bind-to-js-function#trick-2-polymorphic-variant--unwrap), but that only works for functions. Do you have any tips on how I could proceed?  

Here is the commit: https://github.com/vloth/res-msw/commit/fd4e632d1aa645f4002cf6930cfaf3a860a8b46c  

Though, the tests are failing now because request.body will always be `PlainText` . 
Thanks in advance!

**Comment by vloth** [comment]
Update:
I went with a slightly different approach, where I created a `getBody` function to return the body of the request in the correct type variant. It works not, but not happy with the solution and the `Obj.magic`. If there is a way to have the bindings done automatically by rescript I would prefer that.

(link to the update: https://github.com/vloth/res-msw/commit/30e8d3f293bbfa22bfa1049ac9b76b146600290a)

Any input is appreciated!

**Comment by jihchi** [comment]
Hey @vloth , 

Thanks for the update and spending time on it!

I've gotten another idea which hopefully can make user life easier, I've committed the code to a branch for you reference: https://github.com/jihchi/res-msw/pull/17/files

The idea is defined the type of `body` using [Type Parameter](https://rescript-lang.org/docs/manual/latest/type#type-parameter-aka-generic) and wrap it in `Js.Undefined.t`, looks like:

```res
type request<'body> = {body: Js.Undefined.t<'body>, params: Js.Dict.t<string>}
```

It is more flexible than variant since users won't have to pattern match the body to "extract / unwrap" its value.

**Comment by vloth** [comment]
Hi @jihchi,

That is fantastic! Thanks for being so patient and for guiding me through the changes! I learned a lot in the way. 
I just extended the work you initiated to include the tests here: https://github.com/vloth/res-msw/commit/3006890684a25671bccb81da063fee8922f6d8ea. 
Happy to create a PR if that's of your liking! Thanks again!

**Comment by jihchi** [comment]
It looks good to me! PR is very welcome!

**Comment by jihchi** [comment]
Fixes in #19
